Rubberduck.RibbonDispatcher/AbstractRibbonDispatcher/RibbonFactory.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.Office.Core;
+
+namespace RubberDuck.RibbonDispatcher {
+    using System;
+    using static RibbonControlSize;
+    using System.Runtime.InteropServices;
+
+    using ControlSize         = RibbonControlSize;
+    using LanguageStrings     = IRibbonTextLanguageControl;
+
+    [ComVisible(true)][CLSCompliant(true)]
+    public class RibbonFactory {
+        internal RibbonFactory(IRibbonUI ribbonUI) {
+            RibbonUI   = ribbonUI;
+
+            _Controls  = new Dictionary<string,IRibbonCommon>();
+            _Buttons   = new Dictionary<string,IRibbonButton>();
+            _Toggles   = new Dictionary<string,IRibbonToggle>();
+            _DropDowns = new Dictionary<string,IRibbonDropDown>();
+        }
+
+        private IRibbonUI    RibbonUI    { get; }
```

Comment:
Should be a field?

Reply:
Maybe - Thought this might get exposed as I wrote it.